File: Java/src/com/dbc/algorithm/Pointer/RemoveDuplicatesFromSortedArray.java
package com.dbc.algorithm.Pointer;

public class RemoveDuplicatesFromSortedArray {
    public int removeDuplicates(int[] nums) {
        int length = nums.length;
        if (length <= 1) return length;
        int slow = 1, fast = 1;
        while (fast < length) {
            if (nums[fast] != nums[slow - 1]) {
                nums[slow] = nums[fast];
                slow++;
            }
            fast++;
        }
        return slow;
    }
}

File: Python/pointer/remove_duplicates_from_sorted_array.py
from typing import *

def removeDuplicates(nums: List[int]) -> int:
    length = len(nums)
    if length <= 1: return length
    slow, fast = 1, 1
    while fast < length:
        if nums[fast] != nums[slow - 1]:
            nums[slow] = nums[fast]
            slow += 1
        fast += 1
    
    return slow